openeo/rest/datacube.py

```diff
@@ -2152,10 +2149,8 @@ def download(
             (overruling the connection's ``auto_validate`` setting).
         :return: None if the result is stored to disk, or a bytes object returned by the backend.
         """
-        if format is None and outputfile:
-            # TODO #401/#449 don't guess/override format if there is already a save_result with format?
-            format = guess_format(outputfile)
-        cube = self._ensure_save_result(format=format, options=options)
+        weak_format = guess_format(outputfile) if outputfile else None
+        cube = self._ensure_save_result(format=format, options=options, weak_format=weak_format, method="Download")
```

Comment:
`execute_batch` also uses `guess_format` to get this "weak" format indicator and should get same treatment